### PR TITLE
gh-124127: Make Py_REFCNT() opaque in limited C API 3.14

### PR DESCRIPTION
**Pull request by vstinner**
* Issue: gh-124127

----
📚 Documentation preview 📚: https://cpython-previews--124128.org.readthedocs.build/

### CODE REVIEW

**Comment by vstinner** [comment]
cc @colesbury @Fidget-Spinner @encukou

**Comment by vstinner** [comment]
cc @serhiy-storchaka

**Comment by vstinner** [comment]
I plan to merge this change at the end of the week, unless someone wants to review it or has concerns about it.

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

File: Include/refcount.h [hunk 1]
```diff
@@ -77,21 +77,29 @@ check by comparing the reference count field to the immortality reference count.
...
+    #endif
     }
-    Py_ssize_t shared = _Py_atomic_load_ssize_relaxed(&ob->ob_ref_shared);
-    return _Py_STATIC_CAST(Py_ssize_t, local) +
-           Py_ARITHMETIC_RIGHT_SHIFT(Py_ssize_t, shared, _Py_REF_SHARED_SHIFT);
-#endif
-}
-#if !defined(Py_LIMITED_API) || Py_LIMITED_API+0 < 0x030b0000
-#  define Py_REFCNT(ob) Py_REFCNT(_PyObject_CAST(ob))
+    #if !defined(Py_LIMITED_API) || Py_LIMITED_API+0 < 0x030b0000
```
Comment (encukou) [inline Include/refcount.h:100]: This `#if` is now redundant, right?

**Review by encukou — APPROVED** [review]
+1, with a nitpick on the implementation:

File: Include/refcount.h (hunk 1 above)
Comment (vstinner) [reply Include/refcount.h:100]: It's still needed. (Correct me if I'm wrong.)

**Comment by vstinner** [comment]
Merged, thanks for reviews.

**Pull request merged**